src/iptvsimple/data/EpgEntry.cpp

```diff
@@ -220,15 +220,20 @@ bool EpgEntry::UpdateFrom(const xml_node& channelNode, const std::string& id,
 
   m_genreString = GetJoinedNodeValues(channelNode, "category");
 
-  const std::string dateString = GetNodeValue(channelNode, "date");
+  std::string dateString = GetNodeValue(channelNode, "date");
   if (!dateString.empty())
   {
     static const std::regex dateRegex("^[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]");
     if (std::regex_match(dateString, dateRegex))
     {
-      m_firstAired = ParseAsW3CDateString(dateString);
-      // Always compare to the raw string date value to avoid timezone issues
-      if (m_firstAired == ParseAsW3CDateString(strStart))
+      dateString = dateString.substr(0,8) + strStart.substr(8);
+      long long tmpDate = ParseDateTime(dateString);
+      // Windows localtime_s doesnt support negative time_t
+      if tmpDate < 0
+        m_firstAired = ParseAsW3CDateString(dateString);
+      else
+        m_firstAired = ParseAsW3CDateString(static_cast<time_t>(tmpDate));
```

Comment:
You could just embed the ParseDateTime(dateString) call here, no need for tempDate var

Reply:
we need tmpDate to check if its before 0

Reply:
Opps, so we do.